app/interactive_prompt.py

```diff
@@ -171,6 +172,38 @@ def noOp(data):
     def makeCommand(self, cmdLine, view):
         return {}, u'making stuff'
 
+    def openCommand(self, cmdLine, view):
+        """
+        Opens the file under cursor.
+        """
+        penLine = view.textBuffer.lines[view.textBuffer.penRow]
+        wordStart = wordEnd = view.textBuffer.penCol
+
+        # Locate the boundaries of the word under the cursor
+        while wordStart > 0 and penLine[wordStart - 1] != ' ':
+            wordStart -= 1
+
+        penLineLength = len(penLine)
+
+        while wordEnd < penLineLength and penLine[wordEnd] != ' ':
+            wordEnd += 1
+
+        selectedWord = penLine[wordStart: wordEnd]
+        expandedPath = os.path.abspath(os.path.expanduser(selectedWord))
+
+        if not os.path.exists(expandedPath) or not os.path.isfile(expandedPath):
```

Comment:
Will this cover the current cases and check read access?
if not os.access(expandedPath, os.W_OK) or not os.path.isfile ...
The case I'm concerned about is a file that exists, but we don't have read permission for (chmod -w foo).

Reply:
Agree, will modify